src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -507,18 +515,17 @@ FinalState finalState
                             => finalState,
 
                         // We've got nothing.  Build it from scratch :(
-                        NoCompilationState
+                        NoCompilationState noCompilationState
                             => await BuildFinalStateFromScratchAsync(
                                 compilationState,
-                                state.GeneratorInfo,
+                                noCompilationState,
                                 cancellationToken).ConfigureAwait(false),
 
                         // We have a declaration compilation, use it to reconstruct the final compilation
                         AllSyntaxTreesParsedState allSyntaxTreesParsedState
                             => await FinalizeCompilationAsync(
                                 compilationState,
-                                allSyntaxTreesParsedState.CompilationWithoutGeneratedDocuments,
-                                allSyntaxTreesParsedState.GeneratorInfo,
+                                allSyntaxTreesParsedState,
                                 compilationWithStaleGeneratedTrees: null,
```

Comment:
note: open question if we should pass allSyntaxTreesParsedState.CompilationWithGeneratedDocuments here to help pass along useful prior data that could speed things up.  i'll do a separate pr for that though.